js/main.js

```diff
@@ -0,0 +1,172 @@
+];
+
+function redraw() {
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  // console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/** @record */
+class GameBoard {
+  constructor() {
+    /** @type {number} */
+    this.startPointX;
+    /** @type {number} */
+    this.startPointY;
+    /** @type {number} */
+    this.gameHeight;
+    /** @type {number} */
+    this.gameWidth;
+    /** @type {number} */
+    this.dimensionsOfSquare;
+    /** @type {!Array<!Array<!Square>>} */
+    this.board;
+  }
+}
+
+/** @record */
+class Square {
+  constructor() {
+    /** @type {number} */
+    this.xCoord;
+    /** @type {number} */
+    this.yCoord;
+    /** @type {string} */
+    this.character;
+  }
+}
+
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {!GameBoard}
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let /** @type {!GameBoard} */ gameBoard = {
+    startPointX: 0,
+    startPointY: 0,
+    gameHeight: 10,
+    gameWidth: 10,
+    dimensionsOfSquare: 0,
+    board: []
+  }
+
+  let heightScaler = canvasHeight / gameBoard.gameHeight;
+  let widthScaler = canvasWidth / gameBoard.gameWidth;
+
+  gameBoard.dimensionsOfSquare =
+      widthScaler > heightScaler ? heightScaler : widthScaler
+
+  let middlePoint = canvasWidth / 2 -
+      (gameBoard.dimensionsOfSquare * gameBoard.gameWidth / 2);
+
+  gameBoard.startPointX = middlePoint;
+
+  let currX = gameBoard.startPointX;
+  let currY = gameBoard.startPointY;
+
+  for (let i = 0; i < gameBoard.gameHeight; ++i) {
+    let rowOfSquares = [];
+    for (let j = 0; j < gameBoard.gameWidth; ++j) {
+      let square = {xCoord: currX, yCoord: currY, character: '-'};
+
+      rowOfSquares.push(square);
+
+      currX += gameBoard.dimensionsOfSquare;
+    }
+    gameBoard.board.push(rowOfSquares);
+    currX = gameBoard.startPointX;
+    currY += gameBoard.dimensionsOfSquare;
+  }
+
+  return gameBoard;
+}
+
+/**
+ * Converts ranNumber from a float to a character in the
+ * english alphabet based on what interval it is within the charFrequency.
+ * @param {number} ranNumber a random float between 0-101.432
+ * @return {string}
+ */
+function getCharater(ranNumber) {
+  let index = 0;
+
+  for (let i = 0; i < 26; ++i) {
+    if (ranNumber < LETTER_FREQUENCY[i]) {
+      index = i;
+      break;
+    }
+  }
+
+  return String.fromCharCode(97 + index)
+}
+
+/**
+ * Based on a weighted random distribution, it populates the wordSearchBoard
+ * with random letters.
+ * @param {!GameBoard}
+ *     wordSearchBoard
+ */
+function randomizeCharacters(wordSearchBoard) {
+  for (const row of wordSearchBoard.board) {
+    for (const /** @const {!Square} */ square of row) {
```

Comment:
You shouldn't need type annotation here. Compiler should be smart enough to know that row is `Array<!Square>` so square is `!Square`.

Reply:
Updated.